crates/turbopack-ecmascript/src/lib.rs

```diff
@@ -428,13 +446,20 @@ async fn gen_content(
 
     let mut code_gens = Vec::new();
     for r in references.await?.iter() {
-        if let Some(code_gen) = CodeGenerateableVc::resolve_from(r).await? {
+        let r = r.resolve().await?;
+        if let Some(code_gen) = CodeGenerateableWithAvailabilityInfoVc::resolve_from(r).await? {
+            code_gens.push(code_gen.code_generation(context, availability_info));
+        } else if let Some(code_gen) = CodeGenerateableVc::resolve_from(r).await? {
             code_gens.push(code_gen.code_generation(context));
         }
     }
     for c in code_generation.await?.iter() {
         let c = c.resolve().await?;
-        code_gens.push(c.code_generation(context));
+        if let Some(code_gen) = CodeGenerateableWithAvailabilityInfoVc::resolve_from(c).await? {
```

Comment:
It's weird that we have `CodeGenerateable` implementors that will actually never have their `code_generation` implementation called. Perhaps `code_generation` should be an enum of `Default(CodeGenerateableVc), WithAvailabilityInfo(CodeGenerateableWithAvailabilityInfoVc)`.

Reply:
Ok sounds good. That allows to save one resolve operation.